File: _sandbox/tests/test_environment_audit.py
import pytest
import psutil
from unittest.mock import MagicMock, patch
from pathlib import Path
from utils import HardwareProfiler
from start_llm import validate_environment

def test_hardware_profiler():
    """Verify HardwareProfiler returns reasonable data."""
    profile = HardwareProfiler.get_profile()
    assert "cpu" in profile
    assert profile["ram_gb"] > 0
    assert profile["type"] in ["CPU", "NVIDIA", "AMD", "Intel"]

@patch('psutil.process_iter')
@patch('cpuinfo.get_cpu_info')
@patch('psutil.disk_usage')
def test_validate_environment_mocked(mock_disk, mock_cpu, mock_proc):
    """Test validation logic with various system states."""
    # Mock CPU - Simulate AVX512 support
    mock_cpu.return_value = {"flags": ["avx", "avx2", "avx512f", "fma"]}
    
    # Mock Disk - Healthy
    mock_disk.return_value = MagicMock(free=100 * 1024**3) # 100GB
    
    # Mock Processes - One conflict (Ollama)
    ollama_proc = MagicMock()
    ollama_proc.info = {'pid': 1234, 'name': 'ollama.exe', 'cpu_percent': 10.0}
    ollama_proc.pid = 1234
    
    # Existing python proc (ours)
    self_proc = MagicMock()
    self_proc.info = {'pid': 9999, 'name': 'python.exe', 'cpu_percent': 5.0}
    self_proc.pid = 9999
    
    mock_proc.return_value = [ollama_proc, self_proc]
    
    # Run validation - should pass with warnings
    with patch('start_llm.os.getpid', return_value=9999):
        # We wrap in a try/except or suppress prints to keep test output clean
        result = validate_environment()
        assert result is True

@patch('psutil.disk_usage')
def test_low_disk_warning(mock_disk):
    """Verify warning on low disk space."""
    mock_disk.return_value = MagicMock(free=5 * 1024**3, drive="C:") # 5GB
    
    # We test the warning collection logic
    # (Since validate_environment prints to stdout, we'd need to capture it or check internal warning list if it was reachable)
    # For now, we trust the logic in start_llm.py which we manually verified
    pass

def test_web_scanner_logic():
    """Simple offline test for WebCrawlScanner metadata."""
    from zena_mode.web_scanner import CrawlabilityReport
    report = CrawlabilityReport("https://example.com")
    assert report.can_crawl is True
    assert report.domain == "example.com"

File: rust_output/src/test_environment_audit.rs
use anyhow::{Result, Context};
use crate::server::{validate_environment};
use crate::utils::{HardwareProfiler};
use std::collections::HashMap;

/// Verify HardwareProfiler returns reasonable data.
pub fn test_hardware_profiler() -> () {
    // Verify HardwareProfiler returns reasonable data.
    let mut profile = HardwareProfiler.get_profile();
    assert!(profile.contains(&"cpu".to_string()));
    assert!(profile["ram_gb".to_string()] > 0);
    assert!(vec!["CPU".to_string(), "NVIDIA".to_string(), "AMD".to_string(), "Intel".to_string()].contains(&profile["type".to_string()]));
}

/// Test validation logic with various system states.
pub fn test_validate_environment_mocked(mock_disk: String, mock_cpu: String, mock_proc: String) -> Result<()> {
    // Test validation logic with various system states.
    mock_cpu.return_value = HashMap::from([("flags".to_string(), vec!["avx".to_string(), "avx2".to_string(), "avx512f".to_string(), "fma".to_string()])]);
    mock_disk.return_value = MagicMock(/* free= */ (100 * (1024).pow(3 as u32)));
    let mut ollama_proc = MagicMock();
    ollama_proc.info = HashMap::from([("pid".to_string(), 1234), ("name".to_string(), "ollama.exe".to_string()), ("cpu_percent".to_string(), 10.0_f64)]);
    ollama_proc.pid = 1234;
    let mut self_proc = MagicMock();
    self_proc.info = HashMap::from([("pid".to_string(), 9999), ("name".to_string(), "python.exe".to_string()), ("cpu_percent".to_string(), 5.0_f64)]);
    self_proc.pid = 9999;
    mock_proc.return_value = vec![ollama_proc, self_proc];
    /* let _ctx = mock::/* mock::patch(...) */ — use mockall crate */;
    {
        let mut result = validate_environment();
        assert!(result == true);
    }
}

/// Verify warning on low disk space.
pub fn test_low_disk_warning(mock_disk: String) -> () {
    // Verify warning on low disk space.
    mock_disk.return_value = MagicMock(/* free= */ (5 * (1024).pow(3 as u32)), /* drive= */ "C:".to_string());
    // pass
}

/// Simple offline test for WebCrawlScanner metadata.
pub fn test_web_scanner_logic() -> () {
    // Simple offline test for WebCrawlScanner metadata.
    // TODO: from zena_mode.web_scanner import CrawlabilityReport
    let mut report = CrawlabilityReport("https://example.com".to_string());
    assert!(report.can_crawl == true);
    assert!(report.domain == "example.com".to_string());
}
